src/main/java/com/puppycrawl/tools/checkstyle/utils/TokenUtil.java

```diff
@@ -353,4 +353,20 @@ public static BitSet asBitSet(String... tokens) {
                 .collect(BitSet::new, BitSet::set, BitSet::or);
     }
 
+    /**
+     * Check whether given node a preceding access modifier or not.
+     *
+     * @param node the node whose predecessors are to be checked.
+     * @return true if node is preceded by an access modifier else false.
+     */
+    public static boolean isPrecededByAnyModifier(DetailAST node) {
```

Comment:
please move it to src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java
and test it by `InputIndentationxxxxxxx.java` not a pure junit.